app/scripts/modules/netflix/feedback/feedback.directive.ts

```diff
@@ -0,0 +1,61 @@
+import { module } from 'angular';
+import {IModalService} from 'angular-ui-bootstrap';
+
+import { DirectiveFactory } from 'core/utils/tsDecorators/directiveFactoryDecorator';
+
+import './feedback.less';
+
+
+export class FeedbackController implements ng.IComponentController {
+  static get $inject(): string[] {
+    return ['$location', '$scope', '$uibModal', 'settings'];
+  }
+
+  public constructor (private $location: ng.ILocationService,
+                      private $scope: any,
+                      private $uibModal: IModalService,
+                      private settings: any) {}
+
+  public initialize(): void {
+    this.$scope.slackConfig = this.settings.feedback ? this.settings.feedback.slack : null;
+
+    this.$scope.state = {
+      open: false,
+      isMac: navigator.platform.toLowerCase().includes('mac'),
+    };
+
+    this.$scope.setOpen = (open: boolean) => {
+      this.$scope.state.open = open;
+    };
+
+    this.$scope.getCurrentUrlMessage = () => {
+      return encodeURIComponent('(via ' + this.$location.absUrl() + ')\n');
+    };
+
+    this.$scope.openFeedback = () => {
+      this.$uibModal.open({
+        templateUrl: require('./feedback.modal.html'),
+        controller: 'FeedbackModalCtrl as ctrl'
+      });
+    };
+  }
+}
+
+@DirectiveFactory()
+class FeedbackDirective implements ng.IDirective {
+  public restrict = 'E';
+  public replace = true;
+  public templateUrl = require('./feedback.html');
+  public controller: any = FeedbackController;
+  public controllerAs = '$ctrl';
+
+  link($scope: ng.IScope, _$element: JQuery) {
+    const $ctrl: FeedbackController = $scope['$ctrl'];
+    $ctrl.initialize();
+  }
+}
+
+export const FEEDBACK_DIRECTIVE = 'spinnaker.netflix.feedback.directive';
+module(FEEDBACK_DIRECTIVE, [
+  require('core/config/settings')
+]).directive('feedback', <any>FeedbackDirective);
```

Comment:
Instead of a directive, I would make this a component - we really only want to use directives for things like validation or additive behavior on DOM elements.